syntax = "proto3";

package example;

option cc_generic_services = true;

message LoginRequest
{
    bytes name = 1;
    bytes pwd = 2;
}

message ResultCode 
{
    int32 errcode = 1;
    bytes errmsg = 2;
}

message LoginResponse
{
    ResultCode rc = 1;
    bool success = 2;
}

message RegisterRequest
{
    uint32 id = 1;
    bytes name = 2;
    bytes pwd = 3;
}

message RegisterResponse
{
    ResultCode rc = 1;
    bool success = 2;
}

message GetFriendListRequest
{
    uint32 userid = 1;
}

message getFriendListResponse
{
    ResultCode rc  = 1;
    repeated bytes friends = 2;

}

service UserServicesRpc
{
    // 函数名为Login
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc Register(RegisterRequest) returns(RegisterResponse);
}

service GetFriendListSerivceRpc
{
    rpc getFriendList(GetFriendListRequest) returns(getFriendListResponse);
}